services/soundcloud/SoundcloudChannelUrlIdHandler.java

```diff
@@ -68,14 +69,8 @@ public String cleanUrl(String siteUrl) throws ParsingException {
 
     @Override
     public boolean acceptUrl(String channelUrl) {
-        return channelUrl.startsWith("https://soundcloud.com/")
-                && (channelUrl.split("/").length == 4
-                || channelUrl.endsWith("/tracks")
-                || channelUrl.endsWith("/albums")
-                || channelUrl.endsWith("/sets")
-                || channelUrl.endsWith("/reposts")
-                || channelUrl.endsWith("/followers")
-                || channelUrl.endsWith("/following"));
+        String regex = "^https?://(www\\.)?soundcloud.com/[0-9a-z_-]+(/((tracks|albums|sets|reposts|followers|following)/?)?)?([#?].*)?$";
```

Comment:
There is no need to accept non-HTTPS URLs, is there? I strongly doubt Soundcloud themselves redirect to non-HTTPS any more.